sympy/solvers/ode/tests/test_ode.py

```diff
@@ -32,6 +32,160 @@
 # constant_renumber because it will normalize it (constant_renumber causes
 # dsolve() to return different results on different machines)
 
+def _test_for_particular_hint(our_hint, ode_example):
+    if our_hint in classify_ode(ode_example):
+        dsolve_sol = dsolve(ode_example, hint=our_hint)
+        expected_checkodesol = [(True, 0) for i in range(len(dsolve_sol))]
+        return checkodesol(ode_example, dsolve_sol) != expected_checkodesol
+    return False
+
+
+def _ode_solver_test(ode_examples):
+    our_hint = ode_examples['hint']
+    for example in ode_examples['examples']:
+        eq = ode_examples['examples'][example]['eq']
+        sol = ode_examples['examples'][example]['sol']
+        if our_hint not in classify_ode(eq):
+            message = """\
+Hint did not match the example {example}.
```

Comment:
The control flow in this function would be a lot easier to follow if you move these large strings out of the function and have them as module globals.